src/interfaces/chain.cpp

```diff
@@ -234,6 +227,12 @@ class ChainImpl : public Chain
         }
         return nullopt;
     }
+    uint256 getBlockHash(int height) override
+    {
+        LOCK(::cs_main);
+        CBlockIndex* block = ::ChainActive()[height];
+        return block->GetBlockHash();
```

Comment:
why remove the assert and make it undefined behavior?

27b1af965e

Reply:
re: https://github.com/bitcoin/bitcoin/pull/16426#discussion_r417366808

> why remove the assert and make it undefined behavior?
> 
> [27b1af9](https://github.com/bitcoin/bitcoin/commit/27b1af965ed9272496787663ea2869f05c2e5342)

I think it came from a suggestion I made https://github.com/bitcoin/bitcoin/pull/16426#discussion_r412493917 that wasn't clear  about a desire not to crash :smile:. But this is only called one place where it's extremely unlikely to trigger (there'd have to be a very specific thread pre-emption at the same time as a reorg at the same time as loading a wallet)

https://github.com/ariard/bitcoin/blob/49ecec42ea81e9cbf3970416ead79e18863dea87/src/wallet/wallet.cpp#L3927-L3929

Also this method should be removed with #15719

Reply:
> But this is only called one place where it's extremely unlikely to trigger

Sounds like an excellent use-case for `assert` instead of extremely unlikely undefined behavior. Especially given that the assert was there in the fist place.

I don't see why removing the assert can simply wait until #15719

Reply:
Agree with @MarcoFalke here.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/16426#discussion_r417555069

> I don't see why removing the assert can simply wait until #15719

Of course. My suggestion was never to remove the assert by segfaulting instead. Originally the suggestion was to return Optional https://github.com/bitcoin/bitcoin/pull/16426#discussion_r357761094. Just pointing out what's done here isn't too important because it's basically impossible to hit this condition and there's already a PR deleting the method

Reply:
By removing `cs_main` lock taking you may have now a block nullptr in the case where between `getHeight`/`findFirstBlockWithTimeAndHeight` and `getBlockHash` calls you have a thread taking `cs_main` lock, a reorg happening while at same time wallet loading. It's really unlikely but glad #15719 is cleaning this as a follow-up.